src/averageAges.js

```diff
@@ -37,9 +37,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredPeople = withChildren
+    ? people.filter((person, id, peopleArr) => peopleArr.some((candidate) =>
+      candidate.mother === person.name))
+    : people;
+
+  return calculateAverageAge(filteredPeople, 'f');
 }
 
+const calculateAverageAge = (people, gender) => {
+  let filteredPeople = people.filter((person) => person.sex === gender);
+
+  filteredPeople = filteredPeople.map((person) => person.died - person.born);
+
+  return filteredPeople.reduce((person1, person2) =>
```

Comment:
CL2: [CODE STYLE] - If you move your shortened return statement in arrow function to a newline, use round brackets